src/web/common/components/Form/NumberInput.tsx

```diff
@@ -24,10 +24,9 @@ export const NumberInput = ({ name, min, max }: Props) => {
         {...field}
         min={min}
         max={max}
-        value={field.value as number} // should be able to type-safe this but seems hard and not worth effort
+        value={field.value as number}
         onChange={(event, value) => {
           event.preventDefault(); // don't trigger default form submit, which would refresh the page
-          if (value === null) throw new Error("NumberInput should not allow empty value");
 
```

Comment:
I think the changes in this file should be undone, unless you have particular reason for them.

The comment about type-safing still seems relevant to convey that the only reason it's not type-safe is because we didn't take the time for it.

The error is being thrown to be explicit that we've chosen _not_ to handle a `null` `value`.